autoload/OmniSharp/actions/diagnostics.vim

```diff
@@ -56,7 +56,41 @@ endfunction
 
 function! s:StdioCheckRH(Callback, response) abort
   if !a:response.Success | return | endif
-  call a:Callback(OmniSharp#locations#Parse(a:response.Body.QuickFixes))
+  call a:Callback(OmniSharp#locations#Parse(a:response.Body.QuickFixes,
+        \ function("s:DiagnosticQuickfixFixup")))
+endfunction
+
+function! s:DiagnosticQuickfixFixup(quickfix) abort
+  let exclude_paths = get(g:, 'OmniSharp_diagnostic_exclude_paths', [])
+  if len(exclude_paths) && has_key(a:quickfix, 'FileName')
+    for exclude_path in exclude_paths
+      if match(a:quickfix.FileName, exclude_path) > 0
+        return
+      endif
+    endfor
+  endif
+
+  let loglevel = get(a:quickfix, 'LogLevel', '')
+  if loglevel !=# ''
```

Comment:
This should probably just be `has_key(a:quickfix, 'LogLevel')` since it's not used anywhere.

Reply:
Actually.. can this be removed altogether?  @nickspoons

Reply:
Actually I think everything from [locations.vim 57-79](https://github.com/OmniSharp/omnisharp-vim/pull/606/files#diff-de967f1765544fcf25b39ee9228a0b91R57-R79) can be moved in here - then we don't need loglevel there but we do need it for diagnostics.